### PR TITLE
[br-tests] enable BR tests with TREL

### PR DESCRIPTION
**Pull request by simonlingoogle**
This commit runs all Border Router (OTBR) tests with TREL enabled.

### CODE REVIEW

File: script/test [hunk 1]
```diff
@@ -277,12 +278,18 @@ do_build_otbr_docker()
...
         "-DOT_FULL_LOGS=ON"
         "-DOT_UPTIME=ON"
         "-DOTBR_DUA_ROUTING=ON"
         "-DCMAKE_CXX_FLAGS='-DOPENTHREAD_CONFIG_DNSSD_SERVER_BIND_UNSPECIFIED_NETIF=1'"
     )
+
+    if [[ ${TREL} == 1 ]]; then
+        otbr_options+=("-DOT_TREL=ON")
+    else
+        otbr_options+=("-DOT_TREL=OFF")
```
Comment (abtink) [inline script/test:290]: Does the BR test use OT `posix` platform build with a simulated RCP or does it use the simulation platform?

If it is using `posix` platform then I think we may need to set the "trel" interface name as part of the radio URL (I think default it is empty string which practically disables the trel).

**Review by abtink — APPROVED** [review]
LGTM. Thanks.

One question below on whether we may need to set the trel netif name (if using posix).

File: script/test (hunk 1 above)
Comment (simonlingoogle) [reply script/test:290]: > Does the BR test use OT posix platform build with a simulated RCP or does it use the simulation platform?
BR test cases use OT `posix` platform, more specifically `ot-br-posix` Docker image, with a simulated RCP. 

> If it is using posix platform then I think we may need to set the "trel" interface name as part of the radio URL (I think default it is empty string which practically disables the trel).

Actually, the default is `trel://@OTBR_INFRA_IF_NAME@` in OTBR after setup. 
So, we are always giving the TREL URL argument to `otbr-agent` in OTBR Docker. But if `TREL` is not `ON`, the TREL URL argument is not effective at all. 
Thus, we only need to turn `TREL` on to enable `TREL` on these tests.

**Comment by simonlingoogle** [comment]
@jwhui Tip on `Some checks haven’t completed yet`

I think it's expected since the test names are changed by this PR.

File: script/test (hunk 1 above)
Comment (abtink) [reply script/test:290]: Perfect. Thanks 👍

**Review by jwhui — APPROVED** [review]

**Pull request merged**